crates/bevy_hierarchy/src/hierarchy.rs

```diff
@@ -275,9 +271,9 @@ mod tests {
         results.sort_unstable_by_key(|(_, index)| *index);
 
         {
-            let children = world.get::<Children>(grandparent_entity).unwrap();
+            let children = world.get::<Children>(grandparent_entity);
             assert!(
-                !children.iter().any(|&i| i == parent_entity),
+                children.is_none(),
                 "grandparent should no longer know about its child which has been removed"
             );
         }
```

Comment:
I want to explicitly call out that I believe this test was _wrong_ before this PR, since it left an empty `Children` component on the `grandparent` entity.